feat(core, lsp): disable dependency graph

ematipico opened the pull request:
<!--
	Thanks for submitting a pull request!

	We appreciate you spending the time to work on these changes. Please provide enough information so that others can review your pull request.

	Once created, your PR will be automatically labeled according to changed files.

	Before submitting a pull request, please make sure the following is done:

	1. Format and resolve any lint errors: `./rome check --apply`
	2. Verify that tests pass: `./rome test`
	3. Ensure there are no TypeScript errors: `./node_modules/.bin/tsc`

	Learn more about contributing: https://github.com/rome/tools/blob/main/CONTRIBUTING.md
-->

## Summary

Part of #1570 , this PR temporary disables the check of dependencies during `rome check`.

This is a temporary change and it will be turned on again under a flag in the next release, after we figure out what's the correct flow for users.

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->

<!-- Link any relevant issues if necessary or include a transcript of any Discord discussion. -->

## Test Plan

Existing CI

<!-- Demonstrate the code is solid. Example: The exact commands you ran and their output. -->

yassere commented:
I don't think that all this code needs to be commented out. I think it would be better if we added a flag that gets passed into `CheckerOptions` that determines whether or not the dependency analysis would run, and we could have it off by default. That would make it easier for us to test everything when we're working on adding dependency analysis back in.

ematipico commented:
> I don't think that all this code needs to be commented out. I think it would be better if we added a flag that gets passed into `CheckerOptions` that determines whether or not the dependency analysis would run, and we could have it off by default. That would make it easier for us to test everything when we're working on adding dependency analysis back in.

Done! Added a mandatory option and set it `false` every time we create a `Checker`. This change now affects the `LSPServer` too.

yassere commented on internal/core/server/checker/Checker.ts:
```diff
@@ -52,6 +52,7 @@ export type CheckerOptions = {
 	globalDecisions?: LintCompilerOptionsDecisions;
 	lintCompilerOptionsPerFile?: LinterCompilerOptionsPerFile;
 	suppressionExplanation?: string;
+	enableDependenciesCheck: boolean;
```
Could this just be optional? Then it would only need to be included when you do want dependency checking enabled.

ematipico replied:
I intentionally made it required in order to catch all the instances. Let me change it
